### ISSUE TITLE
[VT-100] Vit problem when adding task in russian

### ISSUE THREAD

**Issue by taskwarrior** [scottkosty/vit#122]
_Kai Webber on 2014-08-14T17:33:54Z says:_ 

 When adding task (a) and (!rw task add) in russian I see my chars typed but cursor moves twice - if I type two chars, cursor will move 4 spaces, so end result look like:

Add: ЯЯ   |
when it should be:
Add: ЯЯ|

I can reproduce it in both Ubuntu 14.04 x64 and MacOs Mavericks

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-08-14T17:33:54Z
Modified: 2016-08-14T05:37:02Z
```

**Comment by taskwarrior** [comment]
_Scott Kostyshak on 2014-08-21T08:04:30Z says:_ 

 Thanks for the report Kai and for the simple example. I can reproduce the problem and would at least like to take a look at it before 1.3 is released. I'd be surprised if there weren't other VIT bugs along the same lines. If you come across them, please let us know.

**Comment by taskwarrior** [comment]
_Scott Kostyshak on 2014-08-23T18:20:11Z says:_ 

 The problem is that getch() is called twice for each of the Russian characters. The concatenation of the two separate characters produces the single character output. I tried wgetch() and wget_wch() to try to get a "wide character" but they do not exist. However, they are mentioned (often with words such as "experimental") in /usr/include/curses.h so perhaps there is a way I'm not seeing.

It looks like there is some recent work on this subject. See the section "Wide-Character-Aware Functions" in the following:
http://search.cpan.org/~giraffed/Curses-1.32/Curses.pm

**Comment by taskwarrior** [comment]
_Scott Kostyshak on 2016-08-14T05:37:02Z says:_ 

 Fix pushed to 1.3 branch at 4de7351c. Thanks to the contributions from Richard Gay.
